WordPress/UITestsFoundation/Globals.swift

```diff
@@ -53,6 +53,23 @@ public func waitAndTap( _ element: XCUIElement, maxRetries: Int = 10) {
     }
 }
 
+public func tapUntilExpectedElementAppear( toTap elementToTap: XCUIElement, toAppear elementToAppear: XCUIElement, maxRetries: Int = 10) {
+    var retries = 0
+    while retries < maxRetries {
+        if !elementToAppear.exists {
+            elementToTap.tap()
+            break
+        }
+
+        usleep(500000) // a 0.5 second delay before retrying
+        retries += 1
+    }
+
+    if retries == maxRetries {
+        XCTFail("Expected element (\(elementToAppear)) still does not exists after \(maxRetries) tries.")
```

Comment:
Nit:

```suggestion
        XCTFail("Expected element (\(elementToAppear)) still does not exist after \(maxRetries) taps.")
```

Reply:
this one i'll keep it to `tries`, because it's the number of tries rather than the number of taps (ideally it would just be one tap not many, though we need this because that's not true all the time when the test is running in CI)